common.mk

```diff
@@ -3,22 +3,46 @@
 #########################################################################################
 SHELL=/bin/bash
 
-
 ifndef RISCV
 $(error RISCV is unset. You must set RISCV yourself, or through the Chipyard auto-generated env file)
 else
 $(info Running with RISCV=$(RISCV))
 endif
 
+#########################################################################################
+# specify user-interface variables
+#########################################################################################
+HELP_COMPILATION_VARIABLES += \
+"   EXTRA_GENERATOR_REQS   = requirements needed for the main generator" \
+"   EXTRA_SIM_CFLAGS       = CFLAGS for building simulators" \
+"   EXTRA_SIM_CXXFLAGS     = CXXFLAGS for building simulators" \
+"   EXTRA_SIM_LDFLAGS      = LDFLAGS for building simulators" \
+"   EXTRA_SIM_SOURCES      = simulation sources needed for simulator" \
+"   EXTRA_SIM_REQS         = requirements to build the simulator"
+
+EXTRA_GENERATOR_REQS ?=
+EXTRA_SIM_CXXFLAGS   ?=
+EXTRA_SIM_CFLAGS     ?=
```

Comment:
This variable appears to be unused. I don't think we need separate CXXFLAGS and CFLAGS

Reply:
Additionally, don't the makefiles which edit these variables (ariane,tracegen,nvdla,dromajo) need to be updated?

Reply:
Correct, I remove the `CFLAGS` and just kept `CXXFLAGS`. The other `mk`s don't need to be changed since the names match (except for `dromajo.mk`).